src/PhpSpreadsheet/Cell/DefaultValueBinder.php

```diff
@@ -51,30 +51,43 @@ public static function dataTypeForValue($pValue)
         // Match the value against a few data types
         if ($pValue === null) {
             return DataType::TYPE_NULL;
-        } elseif (is_float($pValue) || is_int($pValue)) {
+        }
+
+        if (is_float($pValue) || is_int($pValue)) {
             return DataType::TYPE_NUMERIC;
-        } elseif (is_bool($pValue)) {
+        }
+
+        if (is_bool($pValue)) {
             return DataType::TYPE_BOOL;
-        } elseif ($pValue === '') {
+        }
+
+        if ($pValue === '') {
             return DataType::TYPE_STRING;
-        } elseif ($pValue instanceof RichText) {
+        }
+
+        if ($pValue instanceof RichText) {
             return DataType::TYPE_INLINE;
-        } elseif (is_string($pValue) && $pValue[0] === '=' && strlen($pValue) > 1) {
+        }
+
+        if ($pValue[0] === '=' && strlen($pValue) > 1) {
             return DataType::TYPE_FORMULA;
-        } elseif (preg_match('/^[\+\-]?(\d+\\.?\d*|\d*\\.?\d+)([Ee][\-\+]?[0-2]?\d{1,3})?$/', $pValue)) {
-            $tValue = ltrim($pValue, '+-');
-            if (is_string($pValue) && $tValue[0] === '0' && strlen($tValue) > 1 && $tValue[1] !== '.') {
-                return DataType::TYPE_STRING;
-            } elseif ((strpos($pValue, '.') === false) && ($pValue > PHP_INT_MAX)) {
+        }
+
+        if ($pValue > PHP_INT_MAX) {
+            if (strpos($pValue, '.') === false) {
                 return DataType::TYPE_STRING;
             }
+        }
+        else if (preg_match('/\.\d*0$/', $pValue)) {
```

Comment:
unnecessary new line.
the `}` should be on the same line as the `else if (`
(see also https://travis-ci.org/PHPOffice/PhpSpreadsheet/jobs/653614524)

Reply:
Fixed.